src/hist/core.py

```diff
@@ -45,24 +45,32 @@ def __call__(self, *args, **kwargs):
 
 @hist.utils.set_family(hist.utils.HIST_FAMILY)
 class BaseHist(bh.Histogram):
+    __slots__ = ("_ax", "_storage_proxy")
+
     def __init__(self, *args, **kwargs):
```

Comment:
The old ordering that I had here was incorrect, and could trip up the `__getattribute__`. By setting the members at the top here, we avoid errors in `__getattribute__`.